app/src/main/java/com/kickstarter/viewmodels/UpdateViewModel.java

```diff
@@ -60,49 +67,81 @@ public ViewModel(final @NonNull Environment environment) {
+        .take(1)
         .filter(ObjectUtils::isNotNull);
 
-      final Observable<String> initialUpdateUrl = update
+      final Observable<Project> project = intent()
+        .map(i -> i.getParcelableExtra(IntentKey.PROJECT))
+        .ofType(Project.class)
+        .filter(ObjectUtils::isNotNull);
+
+      final Observable<String> initialUpdateUrl = initialUpdate
         .map(u -> u.urls().web().update());
 
-      initialUpdateUrl
+      final Observable<String> anotherUpdateUrl = this.goToUpdateRequest
+        .map(request -> request.url().toString());
+
+      Observable.merge(initialUpdateUrl, anotherUpdateUrl)
+        .distinctUntilChanged()
         .compose(bindToLifecycle())
         .subscribe(this.webViewUrl::onNext);
 
-      update
-        .compose(takeWhen(this.shareButtonClickedSubject))
+      final Observable<Update> anotherUpdate = this.goToUpdateRequest
+        .map(this::projectUpdateParams)
+        .switchMap(pu -> this.client.fetchUpdate(pu.first, pu.second).compose(neverError()))
+        .share();
+
+      final Observable<Update> currentUpdate = Observable.merge(initialUpdate, anotherUpdate);
+
+      currentUpdate
+        .compose(takeWhen(this.shareButtonClicked))
         .compose(bindToLifecycle())
         .subscribe(this.startShareIntent::onNext);
 
-      update
-        .compose(takeWhen(this.goToCommentsRequestSubject))
+      currentUpdate
+        .compose(takeWhen(this.goToCommentsRequest))
         .compose(bindToLifecycle())
         .subscribe(this.startCommentsActivity::onNext);
 
-      update
+      currentUpdate
         .map(u -> NumberUtils.format(u.sequence()))
         .compose(bindToLifecycle())
         .subscribe(this.updateSequence::onNext);
 
-      update
-        .compose(takeWhen(this.goToProjectRequestSubject))
-        .switchMap(u ->
-          this.client
-            .fetchProject(String.valueOf(u.projectId()))
-            .compose(neverError())
-        )
+      project
+        .compose(takeWhen(this.goToProjectRequest))
         .compose(bindToLifecycle())
         .subscribe(p -> this.startProjectActivity.onNext(Pair.create(p, RefTag.update())));
+
+      project
+        .compose(takeWhen(this.externalLinkActivated))
+        .compose(bindToLifecycle())
+        .subscribe(p -> this.koala.trackOpenedExternalLink(p, KoalaContext.ExternalLink.PROJECT_UPDATE));
     }
 
-    private final PublishSubject<Request> goToCommentsRequestSubject = PublishSubject.create();
-    private final PublishSubject<Request> goToProjectRequestSubject = PublishSubject.create();
-    private final PublishSubject<Void> shareButtonClickedSubject = PublishSubject.create();
+    /**
+     * Parses a request for project and update params.
+     *
+     * @param request   Comments or update request.
+     * @return          Pair of project param string and update param string.
+     */
+    private @NonNull Pair<String, String> projectUpdateParams(final @NonNull Request request) {
```

Comment:
this method is repeated from `ProjectUpdatesViewModel`, with the same todo of making a better, universal `Navigation` class in the future to handle these requests :/